orchestrator-salt/src/main/resources/salt/salt/sssd/template/add_dns_record.j2

```diff
@@ -17,10 +17,15 @@ FQDN=$(hostname -f)
 IPADDR=$(hostname -i)
 REVERSE_IP=$(hostname -i | awk -F. '{print $4"."$3"." $2"."$1}')
 
-# dnsrecord-add must either add the record or modify it
-if ! ipa dnsrecord-find {{ pillar['sssd-ipa']['domain'] }}. "--name=${HOSTNAME}" "--a-rec=${IPADDR}"  --ttl {{ pillar['sssd-ipa']['dns_ttl'] }} --all; then
-  ipa dnsrecord-add {{ pillar['sssd-ipa']['domain'] }}. "${HOSTNAME}" "--a-rec=${IPADDR}" --ttl {{ pillar['sssd-ipa']['dns_ttl'] }}
-fi
+# add dns a-record 3 times with a random 10-15 seconds interval (see CDPSDX-1981)
+for attempt in {1..3}
+do
+  echo "add dns a-record hostname for ${HOSTNAME}, attempt ${attempt}"
+  if ! ipa dnsrecord-find {{ pillar['sssd-ipa']['domain'] }}. "--name=${HOSTNAME}" "--a-rec=${IPADDR}"  --ttl {{ pillar['sssd-ipa']['dns_ttl'] }} --all; then
+    ipa dnsrecord-add {{ pillar['sssd-ipa']['domain'] }}. "${HOSTNAME}" "--a-rec=${IPADDR}" --ttl {{ pillar['sssd-ipa']['dns_ttl'] }}
+  fi
+  sleep $(((RANDOM % 5) + 10))
```

Comment:
why random? I think we should stick with a fix value

Reply:
Thanks! I changed it to a fixed 10. It is more of a personal habit to put random or exponential in retry as learned from previous experience.